lib/hydra/works/models/concerns/generic_file/characterization.rb

```diff
@@ -0,0 +1,235 @@
+      # will capture the value of the element content
+      @config = {fits_to_rdf_mapping: { filename: "filename",
+                                        file_size: "size",
+                                        well_formed: "well-formed",
+                                        valid: "valid",
+                                        status_message: "message",
+                                        last_modified:  ["lastmodified", "toolname=Exiftool"],
+                                        fits_version: ["fits", "version"],
+                                        exif_version: ["tool", "toolname=Exiftool", "toolversion"],
+                                        original_checksum: "md5checksum" }
+                }.freeze
+
+      # Add attribute methods to the class
+      def self.config=(cfg)
+        @config=cfg
+      end
+
+      def self.config
+        @config
+      end
+
+      # Extract the metadata from a content file and store it as
+      # @param [ActiveFedora::File] file to characterize
+      def characterize(file=original_file)
+        # Run fits and get xml string
+        extracted_md = extract_metadata(file)
+        # Parse xml string to get properties and values
+        prop_values = parse_metadata(extracted_md)
+        # Store metadata as properties of the file.
+        # store_metadata(file.metadata_node, extract_metadata)
+        store_metadata(prop_values)
+      end
+
+
+      protected
+
+      # To make characterization customizealbe, both extract and parse metadata would need to be overridden.
+
+      def extract_metadata(file=original_file)
+        return unless file.has_content?
+        Hydra::FileCharacterization.characterize(file.content, file.original_name, :fits) do |cfg|
+          cfg[:fits] = Hydra::Derivatives.fits_path
+        end
+      end
+
+      # This method could be made to take a SAX Document to enable customization.
+      def parse_metadata(md)
+        parser = Nokogiri::XML::SAX::Parser.new( Hydra::Works::GenericFile::Characterization::SaxDoc.new(config[:fits_to_rdf_mapping]) )
+        parser.parse(md)
+        parser.document.property_values
+      end
+
+      # Assign values of the instance properties from the metadata mapping :prop => val
+      def store_metadata(metadata)
+        metadata.each_pair do |prop, val|
+          if self.respond_to? prop
+            property = self.send(prop)
+            # Multi-valued properties are arrays
+            if property.kind_of? Array
+                property << val
+            else
+              # Single valued property cannot be appended to.
+              if val.kind_of? Array
+                self.send("#{prop}=", val.first)
+              else
+                self.send("#{prop}=", val)
+              end
+            end
+          end
+        end
+      end
+
+    end # of inclduded do
+
+    # SAX parsing class for FITS
+    # Requires fits_to_rdf_mapping
+    class SaxDoc < Nokogiri::XML::SAX::Document
```

Comment:
This inner class needs to be extracted to some other file, but I am not sure what a good destination for it would be.

It's a SAX parser that the characterization concern depends on, so `lib/hydra/works/models/concerns/generic_file/characterization/sax_doc.rb` seems like it should be the place.  That would put a class under the concerns directory tree which seems like a bad plan